File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int n=nums.size();
        if(n==0) return 0;
        int longest=1;
        unordered_set<int> st(nums.begin(),nums.end());
        for(int it:st){
            if(st.find(it-1)==st.end()){
                int current=it;
                int count=1;
                while(st.find(current+1)!=st.end()){
                    current++;
                    count++;
                }
                 longest=max(count,longest);
            }
           
        }
        return longest;
        
    }
};

File: Linked_List/128-longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {

        int maxilen =0;

        HashSet<Integer> st = new HashSet<>();

        for(int x:nums) st.add(x);


        for(int x:nums){


            if(!st.contains(x-1)){

                int count =0;
                int start=x;

                while(st.contains(start)){
                   
                    int rm = start;
                     start++;

                    st.remove(rm);
                    count++;
                }

                 maxilen = Math.max(maxilen,count);
            }
        }

        return maxilen;
        
    }
}